compiler/rustc_pattern_analysis/src/lints.rs

```diff
@@ -33,20 +34,24 @@ pub(crate) struct PatternColumn<'p, 'tcx> {
 
 impl<'p, 'tcx> PatternColumn<'p, 'tcx> {
     pub(crate) fn new(arms: &[MatchArm<'p, 'tcx>]) -> Self {
-        let mut patterns = Vec::with_capacity(arms.len());
+        let patterns = Vec::with_capacity(arms.len());
+        let mut column = PatternColumn { patterns };
         for arm in arms {
-            if arm.pat.is_or_pat() {
-                patterns.extend(arm.pat.flatten_or_pat())
-            } else {
-                patterns.push(arm.pat)
-            }
+            column.expand_and_push(PatOrWild::Pat(arm.pat));
         }
-        Self { patterns }
+        column
     }
-
-    fn is_empty(&self) -> bool {
-        self.patterns.is_empty()
+    fn expand_and_push(&mut self, pat: PatOrWild<'p, RustcMatchCheckCtxt<'p, 'tcx>>) {
```

Comment:
kind of a shame this is duplicated w/ `fn expand_and_push` on `Matrix`.

Reply:
(this one pushes individual patterns and skips wildcards, the `Matrix` one pushes rows and doesn't skip wildcards)

Reply:
oh; then it extra sucks that they are named identically. it would be nice if these methods had more doc comments, lol

Reply:
Sure